bugfix: S3C-3388 network.http.Server.setKeepAliveTimeout()

jonathan-gramain opened the pull request:
Add a helper function to set the keep-alive timeout of the node.js
HTTP server managed by the Server class.

jonathan-gramain committed:
bugfix: S3C-3388 network.http.Server.setKeepAliveTimeout()

Add a helper function to set the keep-alive timeout of the node.js
HTTP server managed by the Server class.

bert-e commented:
# Hello jonathan-gramain,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/arsenal/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/S3C-3388-httpServerKeepAliveTimeoutOption` into
`development/7.4`
* [w/7.8/bugfix/S3C-3388-httpServerKeepAliveTimeoutOption](https://github.com/scality/arsenal/compare/development/7.8...w/7.8/bugfix/S3C-3388-httpServerKeepAliveTimeoutOption) will be merged into `development/7.8` (pull request #1284)
* [w/7.9/bugfix/S3C-3388-httpServerKeepAliveTimeoutOption](https://github.com/scality/arsenal/compare/development/7.9...w/7.9/bugfix/S3C-3388-httpServerKeepAliveTimeoutOption) will be merged into `development/7.9` (pull request #1285)
* [w/8.1/bugfix/S3C-3388-httpServerKeepAliveTimeoutOption](https://github.com/scality/arsenal/compare/development/8.1...w/8.1/bugfix/S3C-3388-httpServerKeepAliveTimeoutOption) will be merged into `development/8.1` (pull request #1286)



The following branches will **NOT** be impacted:

* `development/6.4`
* `stabilization/7.8.0`




*Follow* integration pull requests if you would like to be notified of
build statuses by email.

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers





Peer approvals *must* include a mandatory approval from @jonathan-gramain.

jonathan-gramain committed:
bugfix: S3C-3388 constants for HTTP connection timeouts

Add constants related to HTTP client/server connection timeouts with
values avoiding ECONNRESET errors due to the server closing
connections that clients are attempting to reuse at the same moment.

bert-e commented:
# Conflict




A conflict has been raised during the update of
integration branch `w/7.8/bugfix/S3C-3388-httpServerKeepAliveTimeoutOption` with contents from `bugfix/S3C-3388-httpServerKeepAliveTimeoutOption`
and `development/7.8`.

Please resolve the conflict on **the integration branch** (`w/7.8/bugfix/S3C-3388-httpServerKeepAliveTimeoutOption`).


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout w/7.8/bugfix/S3C-3388-httpServerKeepAliveTimeoutOption
 $ git pull  # or "git reset --hard origin/w/7.8/bugfix/S3C-3388-httpServerKeepAliveTimeoutOption"
 $ git merge origin/development/7.8
 $ # <intense conflict resolution>
 $ git commit
 $ git merge origin/bugfix/S3C-3388-httpServerKeepAliveTimeoutOption
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/7.8/bugfix/S3C-3388-httpServerKeepAliveTimeoutOption
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers





Peer approvals *must* include a mandatory approval from @jonathan-gramain.

bert-e commented:
# History mismatch



Merge commit #5e6a56c888eeb3699bee67e462f5823b7913945e on the integration branch
`w/7.8/bugfix/S3C-3388-httpServerKeepAliveTimeoutOption` is merging a branch which is neither the current
branch `bugfix/S3C-3388-httpServerKeepAliveTimeoutOption` nor the development branch
`development/7.8`.

It is likely due to a rebase of the branch `bugfix/S3C-3388-httpServerKeepAliveTimeoutOption` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

jonathan-gramain commented:
/reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/7.8/bugfix/S3C-3388-httpServerKeepAliveTimeoutOption` with contents from `bugfix/S3C-3388-httpServerKeepAliveTimeoutOption`
and `development/7.8`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/7.8/bugfix/S3C-3388-httpServerKeepAliveTimeoutOption origin/development/7.8
 $ git merge origin/bugfix/S3C-3388-httpServerKeepAliveTimeoutOption
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/7.8/bugfix/S3C-3388-httpServerKeepAliveTimeoutOption
```

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/S3C-3388-httpServerKeepAliveTimeoutOption` into
`development/7.4`
* [w/7.8/bugfix/S3C-3388-httpServerKeepAliveTimeoutOption](https://github.com/scality/arsenal/compare/development/7.8...w/7.8/bugfix/S3C-3388-httpServerKeepAliveTimeoutOption) will be merged into `development/7.8` (pull request #1287)
* [w/7.9/bugfix/S3C-3388-httpServerKeepAliveTimeoutOption](https://github.com/scality/arsenal/compare/development/7.9...w/7.9/bugfix/S3C-3388-httpServerKeepAliveTimeoutOption) will be merged into `development/7.9` (pull request #1288)
* [w/8.1/bugfix/S3C-3388-httpServerKeepAliveTimeoutOption](https://github.com/scality/arsenal/compare/development/8.1...w/8.1/bugfix/S3C-3388-httpServerKeepAliveTimeoutOption) will be merged into `development/8.1` (pull request #1289)



The following branches will **NOT** be impacted:

* `development/6.4`
* `stabilization/7.8.0`




*Follow* integration pull requests if you would like to be notified of
build statuses by email.

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers





Peer approvals *must* include a mandatory approval from @jonathan-gramain.

jonathan-gramain commented:
/approve

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/7.4`

* :heavy_check_mark: `development/7.8`

* :heavy_check_mark: `development/7.9`

* :heavy_check_mark: `development/8.1`



The following branches will **NOT** be impacted:

* `development/6.4`
* `stabilization/7.8.0`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/7.4`

* :heavy_check_mark: `development/7.8`

* :heavy_check_mark: `development/7.9`

* :heavy_check_mark: `development/8.1`



The following branches have **NOT** changed:

* `development/6.4`
* `stabilization/7.8.0`



Please check the status of the associated issue S3C-3388.

Goodbye jonathan-gramain.
